lib/bap_disasm/bap_disasm_reconstructor.ml

```diff
@@ -46,39 +46,68 @@ let is_start roots cfg blk =
   Set.mem roots (Block.addr blk) ||
   Cfg.Node.degree ~dir:`In blk cfg = 0
 
-let entries_of_block cfg roots entries blk =
+let is_fall e = Cfg.Edge.label e = `Fall
+let is_call b = Insn.(is call) (Block.terminator b)
+
+let entries_of_block cfg roots blk =
   let entries =
-    if is_start roots cfg blk then Set.add entries blk
-    else entries in
-  let term = Block.terminator blk in
-  if Insn.(is call) term then
+    if is_start roots cfg blk then Block.Set.singleton blk
+    else Block.Set.empty in
+  if is_call blk then
     Seq.fold ~init:entries (Cfg.Node.outputs blk cfg)
       ~f:(fun entries e ->
-          if Cfg.Edge.label e <> `Fall then
-            Set.add entries (Cfg.Edge.dst e)
-          else entries)
+          if is_fall e then entries
+          else Set.add entries (Cfg.Edge.dst e))
   else entries
 
-let collect_entries cfg roots =
-  let roots = Addr.Set.of_list roots in
-  Seq.fold (Cfg.nodes cfg) ~init:Block.Set.empty
-    ~f:(entries_of_block cfg roots)
+let terminator_addr blk =
```

Comment:
use the block address as a designator of a callsite. A basic block is a unit of a control flow relation not a specific instruction.

Reply:
Is this code dead now? If yes, them remove it, otherwise make sure that it is dead and still remove.